core/types/header_ext.go

```diff
@@ -143,8 +143,11 @@ func (h *HeaderSerializable) updateToExtras(extras *HeaderExtra) {
 	extras.BlockGasCost = h.BlockGasCost
 }
 
-//go:generate go run github.com/fjl/gencodec -type HeaderSerializable -field-override headerMarshaling -out gen_header_json.go
-//go:generate go run github.com/ava-labs/libevm/rlp/rlpgen -type HeaderSerializable -out gen_header_rlp.go
+// NOTE: both generators currently do not support type aliases.
+// We are using custom versions of these programs for now to support type aliases,
```

Comment:
What do you mean by custom versions? These aren't captured by a change to, for example, the `go.sum` entry for `gencodec`.

Reply:
See #833 
For now fjl/gencodec is not pinned to a version at all, and libevm is pinned to the version in go.mod.
I did modify them locally to generate with type aliases, until these are properly merged/released.

Reply:
> I did modify them locally to generate with type aliases, until these are properly merged/released.

Code needs to be reproducible from the repo, not rely on some local changes held on a single developer's machine.

> See https://github.com/ava-labs/coreth/issues/833

LGTM and sufficient to rectify the above as long as it's addressed ASAP.

Reply:
I'll work on it in parallel while next PRs are undergoing reviews, and wrap it up before merging libevm -> master